lib/msf/core/db_manager/session.rb

```diff
@@ -170,54 +115,129 @@ def report_session(opts)
   # @param wspace [Mdm::Workspace]
   # @return [void]
   def infer_vuln_from_session(session, wspace)
-    s = session.db_record
-    host = session.db_record.host
+    ::ActiveRecord::Base.connection_pool.with_connection {
+      s = session.db_record
+      host = session.db_record.host
```

Comment:
If `session.db_record` is already in `s`, then `host` should be `host = s.host` to eliminate a redundant lookup chain.